src/mnode/mnode-controller.cpp

```diff
@@ -450,7 +505,33 @@ CAmount CMasterNodeController::GetArtTicketFeePerKB()
     return ArtTicketFeePerKBDefault;
 }
 
+double CMasterNodeController::GetChainDeflationRate() {
+    if (chainActive.Height() <= ChainBaselineDifficultyUpperIndex + ChainTrailingAverageDifficultyRange) {
+        return ChainDeflationRateDefault;
+    } else {
+        // Get baseline average difficulty
+        double totalBaselineDifficulty = 0.0;
+        for (CAmount i = ChainBaselineDifficultyLowerIndex; i < ChainBaselineDifficultyUpperIndex; i++) {
+            CBlockIndex* index = chainActive[i];
+            totalBaselineDifficulty += getNetworkDifficulty(index, true);
+        }
+        double averageBaselineDifficulty = totalBaselineDifficulty/(ChainBaselineDifficultyUpperIndex - ChainBaselineDifficultyLowerIndex);
+        // Get trailing average difficulty
+        CAmount endTrailingIndex = ChainBaselineDifficultyUpperIndex + ChainTrailingAverageDifficultyRange*((chainActive.Height() - ChainBaselineDifficultyUpperIndex)/ChainTrailingAverageDifficultyRange );
+        CAmount startTrailingIndex = endTrailingIndex - ChainTrailingAverageDifficultyRange;
+        
+        
+        double totalTrailingDifficulty = 0.0;
+        for (CAmount i = startTrailingIndex; i < endTrailingIndex; i++) {
```

Comment:
```
    if (chainActive.Height() <= ChainBaselineDifficultyUpperIndex + ChainTrailingAverageDifficultyRange) {
        return ChainDeflationRateDefault;
```
Else
endTrailingIndex will be somewhere in the middle of ChainBaselineDifficultyUpperIndex and chainActive.Height().
This is because of the calculation recipe that is provide on the freedcamp.